### PR TITLE
Bug correction

### PR DESCRIPTION
**Pull request by tanyasinghal**

### CODE REVIEW

File: app/Http/Controllers/OutgoingLetterLogsController.php [hunk 1]
```diff
@@ -12,15 +12,11 @@ class OutgoingLetterLogsController extends Controller
 {
     public function index(Request $request)
     {
+        $after_date = $request->input('after') ?? OutgoingLetterLog::min('date');
+        $before_date = $request->input('before') ?? OutgoingLetterLog::max('date');
```
Comment (rumansaleem) [inline app/Http/Controllers/OutgoingLetterLogsController.php]: you can directly use `$request->after` and `$request->before` instead of calling `input` method on request.

File: app/Http/Controllers/OutgoingLetterLogsController.php [hunk 2]
```diff
@@ -12,15 +12,11 @@ class OutgoingLetterLogsController extends Controller
...
         $query = OutgoingLetterLog::query();
         
-        if ($request->has('before')) {
-            $query->where('date', '<', $request->before);
-        }
-
-        if ($request->has('after')) {
-            $query->where('date', '>', $request->after);
-        }
+        $query->whereBetween('date',[$after_date,$before_date]);
```
Comment (rumansaleem) [inline app/Http/Controllers/OutgoingLetterLogsController.php]: reduce this to a one liner.

```
$outgoing_letter_logs = OutgoingLetterLog::whereBetween('date', [$after_date, $before_date])->get();
```
then use `compact()` function to pass in view.

File: resources/views/outgoing_letter_logs/index.blade.php [hunk 3]
```diff
@@ -43,7 +43,7 @@ class="form-input is-sm mr-4"
                     <td class="px-3 py-1 border-b">{{ $letter->recipient }}</td>
                     <td class="px-3 py-1 border-b table-fit">{{ $letter->type }}</td>
                     <td class="pl-3 pr-6 py-1 border-b table-fit text-right">
-                        <a href="/outgoing-letter-logs/{{$letter->id}}" 
+                        <a href="/outgoing-letter-logs/{{$letter->id}}/edit"
```
Comment (rumansaleem) [inline resources/views/outgoing_letter_logs/index.blade.php]: you should've made a seperate PR for this one. But for now, leave it as it is.

**Comment by tanyasinghal** [comment]
> Please write a testcase for that empty query parameter bug, and make other changes as suggested.

Do I need to write 3 separate test cases (one where before parameter is empty ,second where after parameter is empty and third where both parameters are empty) or  1 test case (where both parameters are empty)?

**Comment by rumansaleem** [comment]
Writing 3 different test cases is better than having one single testcase with a lot of assertions.
